packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -167,15 +167,13 @@ export class EngineReportingExtension<TContext = any>
       });
     }
 
-    // While clientAddress could be a part of the protobuf, we'll ignore it for
-    // now, since the backend does not group by it and Engine frontend will not
-    // support it in the short term
-    const { clientName, clientVersion } = this.generateClientInfo({
+    const { clientName, clientVersion, clientId } = this.generateClientInfo({
       context: o.context,
       extensions: o.extensions,
     });
     this.trace.clientName = clientName || '';
     this.trace.clientVersion = clientVersion || '';
+    this.trace.clientId = clientId || '';
```

Comment:
@abernix I'm pretty sure this doesn't matter (@glasser could correct me if wrong) since protobuf has no concept of `null` for strings and just uses the empty String. We could probably set the internal value to `null` or `undefined` and the protobuf library would do the mapping, but I'm not positive what the point is.

Reply:
We would also have to change the type of the trace model, which would mean overriding the types generated from protobuf, so I'd prefer not to :)